### ISSUE TITLE
Could not load type IdentityModel.Extensions.HashStringExtensions

### ISSUE THREAD

**Issue by vdaiev** [DuendeArchive/IdentityModel#118]
Hi Guys,

I'm experiencing some issues after libraries update. I have an Owin Katana app with IdentityServer3.
After libraries update experiencing the following behavior. When I'm calling a server to get the token - I get the one. Decoding it on jwt.io shows correct data, roles, scopes etc. When I call an API with the token I get 401 Unauthorized. During the debugging I found that the user is null in the context (no big surprise). No evident errors.

Enabled Katana tracing and got the following message in the log: 
```
Microsoft.Owin.Security.OAuth.OAuthBearerAuthenticationMiddleware Error: 0 : Authentication failed
System.TypeLoadException: Could not load type 'IdentityModel.Extensions.HashStringExtensions' from assembly 'IdentityModel, Version=3.6.1.0, Culture=neutral, PublicKeyToken=null'.
   at IdentityServer3.AccessTokenValidation.ValidationEndpointTokenProvider.<ReceiveAsync>d__1.MoveNext()
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[TStateMachine](TStateMachine& stateMachine)
   at IdentityServer3.AccessTokenValidation.ValidationEndpointTokenProvider.ReceiveAsync(AuthenticationTokenReceiveContext context)
   at Microsoft.Owin.Security.OAuth.OAuthBearerAuthenticationHandler.<AuthenticateCoreAsync>d__0.MoveNext()
    ProcessId=25940
    DateTime=2018-05-31T09:48:51.8158469Z
```

Ok, long story short. Updated the following libraries:
```
IdentityModel, Version=1.0.0.0 -> Version=3.6.1.0
IdentityServer3, Version=2.5.0.0 -> Version=2.6.0.0
Microsoft.Owin, Version=3.0.1.0 -> Version=4.0.0.0
```

and added
```
System.IdentityModel.Tokens.Jwt, Version=4.0.40306.1554
```

I'm sorry in advance if this is some kind of a stupid question or this is not a right place to post such issue

**Comment by leastprivilege** [comment]
The access token validation middleware is not designed to work with IdentityModel 3.x - use the version it references in its nuget definition

https://www.nuget.org/packages/IdentityServer3.AccessTokenValidation/2.15.1.

**Comment by vdaiev** [comment]
Great, thank you!
confirmed, it works with IdentityModel 1.9.2 like a charm
